test/releasers/rust.ts

```diff
@@ -0,0 +1,240 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from 'assert';
+import {describe, it, afterEach} from 'mocha';
+import * as nock from 'nock';
+import {Rust} from '../../src/releasers/rust';
+import {readFileSync} from 'fs';
+import {resolve} from 'path';
+import * as snapshot from 'snap-shot-it';
+import * as suggester from 'code-suggester';
+import * as sinon from 'sinon';
+
+const sandbox = sinon.createSandbox();
+const fixturesPath = './test/releasers/fixtures/rust';
+
+function mockRequest(opts: {monorepo: boolean}) {
+  const snapName = opts.monorepo ? 'monorepo' : 'non-monorepo';
+
+  const crate1Manifest = readFileSync(
+    resolve(fixturesPath, 'Cargo-crate1.toml'),
+    'utf8'
+  );
+  const crate2Manifest = readFileSync(
+    resolve(fixturesPath, 'Cargo-crate2.toml'),
+    'utf8'
+  );
+  const workspaceManifest = readFileSync(
+    resolve(fixturesPath, 'Cargo-workspace.toml'),
+    'utf8'
+  );
+  const graphql = JSON.parse(
+    readFileSync(resolve(fixturesPath, 'commits.json'), 'utf8')
+  );
+
+  let req = nock('https://api.github.com')
```

Comment:
I was finding nock was getting pretty painful, so in some of the newer unit tests I've been mocking methods on the `github` class instead:

https://github.com/googleapis/release-please/blob/master/test/releasers/terraform-module.ts#L101

I have a slight preference for this, but you've already put the footwork in for nock... so I'm open to using your approach.

What are your thoughts?

Reply:
I wasn't familiar with `nock` (it was quite the adventure) but I am familiar with `sinon` so this should be quick! I'm on it.